inbox/util/concurrency.py

```diff
@@ -65,8 +71,17 @@ def wrapped(*args, **kwargs):
 def retry_with_logging(func, logger=None, retry_classes=None,
                        fail_classes=None, account_id=None, provider=None,
                        backoff_delay=BACKOFF_DELAY):
-    def callback():
-        log_uncaught_errors(logger, account_id=account_id, provider=provider)
+    network_errs = [0]
+
+    def callback(e):
+        if isinstance(e, TRANSIENT_NETWORK_ERRS):
+            network_errs[0] += 1
```

Comment:
Are you resetting this anywhere? What happens for example if an IMAP account meets 20 transient SSL errors in the course of a regular run.